src/viper/syntax.ml

```diff
@@ -90,8 +92,21 @@ and stmt' =
 
 and typ = (typ', info) Source.annotated_phrase
 
+
+and mut = Const | Mut
+
 and typ' =
   | IntT
   | BoolT
   | RefT
+  | ArrayT of mut * typ'
```

Comment:
Having thought about this some more, I don't think this is right. This is supposed to be the Viper type, and the `Array` domain defined in `array.vpr` is not parameterized by anything. So this type should be simply `ArrayT` (no `of`)